File: ALODWebUtility/Perms/Search/ItemSelectedEventArgs.vb
Imports ALOD.Core.Domain.Workflow

Namespace Permission.Search

    Public Class ItemSelectedEventArgs
        Inherits EventArgs

        Private _baseType As ModuleType
        Private _canEdit As Boolean

        'used by appeals
        Private _compo As String

        Private _parentId As Integer
        Private _recId As Integer
        Private _redirect As Boolean = True
        Private _refId As Integer
        Private _requestId As Integer
        Private _status As String = ""
        Private _type As ModuleType
        Private _url As String = ""
        Private _workFlowId As Short

        ''' <summary>
        ''' For appeals this is the the base module type being appealed
        ''' For non-appeals this will be the same as Type
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Property BaseType() As ModuleType
            Get
                Return _baseType
            End Get
            Set(ByVal value As ModuleType)
                _baseType = value
            End Set
        End Property

        Public Property CanEdit() As Boolean
            Get
                Return _canEdit
            End Get
            Set(ByVal value As Boolean)
                _canEdit = value
            End Set
        End Property

        Public Property Compo() As String
            Get
                Return _compo
            End Get
            Set(ByVal value As String)
                _compo = value
            End Set
        End Property

        Public Property ParentID() As Integer
            Get
                Return Me._parentId
            End Get
            Set(ByVal value As Integer)
                Me._parentId = value
            End Set
        End Property

        Public Property RecID() As Integer
            Get
                Return Me._recId
            End Get
            Set(ByVal value As Integer)
                Me._recId = value
            End Set
        End Property

        Public Property Redirect() As Boolean
            Get
                Return _redirect
            End Get
            Set(ByVal value As Boolean)
                _redirect = value
            End Set
        End Property

        Public Property RefId() As Integer
            Get
                Return _refId
            End Get
            Set(ByVal value As Integer)
                _refId = value
            End Set
        End Property

        Public Property RequestId() As Integer
            Get
                Return _requestId
            End Get
            Set(ByVal value As Integer)
                _requestId = value
            End Set
        End Property

        Public Property Status() As String
            Get
                Return _status
            End Get
            Set(ByVal value As String)
                _status = value
            End Set
        End Property

        Public Property Type() As ModuleType
            Get
                Return _type
            End Get
            Set(ByVal value As ModuleType)
                _type = value
            End Set
        End Property

        Public Property Url() As String
            Get
                Return _url
            End Get
            Set(ByVal value As String)
                _url = value
            End Set
        End Property

        Public Property WorkFlowId() As Short
            Get
                Return _workFlowId
            End Get
            Set(ByVal value As Short)
                _workFlowId = value
            End Set
        End Property

    End Class

End Namespace

File: ALODWebUtility/Perms/Search/ItemSelectedEventArgs.cs
using System;
using ALOD.Core.Domain.Workflow;

namespace ALODWebUtility.Perms.Search
{
    public class ItemSelectedEventArgs : EventArgs
    {
        private ModuleType _baseType;
        private bool _canEdit;

        // used by appeals
        private string _compo;

        private int _parentId;
        private int _recId;
        private bool _redirect = true;
        private int _refId;
        private int _requestId;
        private string _status = "";
        private ModuleType _type;
        private string _url = "";
        private short _workFlowId;

        /// <summary>
        /// For appeals this is the the base module type being appealed
        /// For non-appeals this will be the same as Type
        /// </summary>
        /// <value></value>
        /// <returns></returns>
        /// <remarks></remarks>
        public ModuleType BaseType
        {
            get
            {
                return _baseType;
            }
            set
            {
                _baseType = value;
            }
        }

        public bool CanEdit
        {
            get
            {
                return _canEdit;
            }
            set
            {
                _canEdit = value;
            }
        }

        public string Compo
        {
            get
            {
                return _compo;
            }
            set
            {
                _compo = value;
            }
        }

        public int ParentID
        {
            get
            {
                return this._parentId;
            }
            set
            {
                this._parentId = value;
            }
        }

        public int RecID
        {
            get
            {
                return this._recId;
            }
            set
            {
                this._recId = value;
            }
        }

        public bool Redirect
        {
            get
            {
                return _redirect;
            }
            set
            {
                _redirect = value;
            }
        }

        public int RefId
        {
            get
            {
                return _refId;
            }
            set
            {
                _refId = value;
            }
        }

        public int RequestId
        {
            get
            {
                return _requestId;
            }
            set
            {
                _requestId = value;
            }
        }

        public string Status
        {
            get
            {
                return _status;
            }
            set
            {
                _status = value;
            }
        }

        public ModuleType Type
        {
            get
            {
                return _type;
            }
            set
            {
                _type = value;
            }
        }

        public string Url
        {
            get
            {
                return _url;
            }
            set
            {
                _url = value;
            }
        }

        public short WorkFlowId
        {
            get
            {
                return _workFlowId;
            }
            set
            {
                _workFlowId = value;
            }
        }
    }
}
